SwordPlugin.inc.php

```diff
@@ -111,23 +134,39 @@ public function callbackAuthorDeposits($hookName, $args) {
 		}
 
 		if ($sendDepositNotification) {
-			$submittingUser = $submission->getPrimaryAuthor();  // TODO how to get submission user ?
-			$contactName = $context->getSetting('contactName');
-			$contactEmail = $context->getSetting('contactEmail');
-			import('classes.mail.ArticleMailTemplate');
-			$mail = new ArticleMailTemplate($submission, 'SWORD_DEPOSIT_NOTIFICATION', null, $context, true);
-			$mail->setFrom($contactEmail, $contactName);
-			$mail->addRecipient($submittingUser->getEmail(), $submittingUser->getFullName());
-
-			$mail->assignParams(array(
-				'journalName' => $context->getLocalizedName(),
-				'articleTitle' => $submission->getLocalizedTitle(),
-				'swordDepositUrl' => $dispatcher->url(
-					$request, ROUTE_PAGE, null, 'sword', 'index', $submission->getId()
-				)
-			));
-
-			$mail->send($request);
+			$submissionAuthors = [];
+			$dao = new StageAssignmentDAO();
+			$daoResult = $dao->getBySubmissionAndRoleId($submission->getId(), ROLE_ID_AUTHOR);
+			while ($record = $daoResult->next()) {
+				$userId = $record->getData('userId');
+				if (!in_array($userId, $submissionAuthors)) {
+					array_push($submissionAuthors, $userId);
+				}
+			}
+
+			$userDao = new UserDao();
```

Comment:
DAOs should always be fetched via `DAORegistry::getDAO`, e.g.:
```
$userDao = DAORegistry::getDAO('UserDAO'); /* @var $userDao UserDAO */
```
(The annotation helps with PHP Scrutinizer and probably some IDEs.)